controllers/harborcluster_controller.go

```diff
@@ -88,7 +89,20 @@ func (r *HarborClusterReconciler) Reconcile(req ctrl.Request) (ctrl.Result, erro
 		}, err
 	}
 
-	harborStatus, err := r.Harbor(ctx, &harborCluster, componentToStatus, nil).Reconcile()
+	getRegistry := func() *string {
```

Comment:
Similar to above comments, please double-check `ImagePullSecret ` whether it is mandatory or not.